agents-common/src/test/java/org/apache/ranger/plugin/policyengine/TestCacheMap.java

```diff
@@ -28,89 +28,79 @@
-				LOG.debug("TestCacheMap.runTests(), Cache Size after insert(): " + testCacheMap.size());
-			}
-		}
-
-		Set<String> keySet = testCacheMap.keySet();
-
-		if (LOG.isDebugEnabled()) {
-			LOG.debug("TestCacheMap.runTests(), KeySet Size:" + keySet.size());
-			LOG.debug("TestCacheMap.runTests(), printing keys..");
-
-		    int i = 0;
-
-		    for (String key : keySet) {
-				LOG.debug("TestCacheMap.runTests(), index:" + i++ + ", key:" + key);
-			}
-
-			LOG.debug("<== TestCacheMap.runTests()");
-		}
-
-	}
+    private static final Logger LOG = LoggerFactory.getLogger(TestCacheMap.class);
+
+    private static       CacheMap<String, String> testCacheMap;
+    private static final int                      initialCapacity = 16;
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        LOG.debug("==> TestCacheMap.setUpBeforeClass(), initialCapacity:{}", initialCapacity);
+
+        testCacheMap = new CacheMap<>(initialCapacity);
+
+        LOG.debug("<== TestCacheMap.setUpBeforeClass(), initialCapacity:{}", initialCapacity);
+    }
+
+    @AfterClass
+    public static void tearDownAfterClass() throws Exception {
+    }
+
+    @Test
+    public void runTests() {
+        LOG.debug("==> TestCacheMap.runTests(), First batch of {} inserts starting from 0", initialCapacity);
+
+        for (int i = 0; i < initialCapacity; i++) {
+            String key   = String.valueOf(i);
+            String value = key;
+
+            LOG.debug("TestCacheMap.runTests(), Inserting into Cache, key:{}, value:{}", key, value);
+
+            testCacheMap.put(key, value);
+
+            LOG.debug("TestCacheMap.runTests(), Cache Size after insert(): {}", testCacheMap.size());
+        }
+
+        LOG.debug("TestCacheMap.runTests(), First batch of {} retrieves counting down from {}", initialCapacity / 2, (initialCapacity / 2 - 1));
+
+        for (int i = initialCapacity / 2 - 1; i >= 0; i--) {
+            String key = String.valueOf(i);
+
+            LOG.debug("TestCacheMap.runTests(), Searching Cache, key:{}", key);
+
+            String value = testCacheMap.get(key);
+
+            if (value == null || !value.equals(key)) {
+                LOG.error("TestCacheMap.runTests(), Did not get correct value for key, key:{}, value:{}", key, value);
+            }
+        }
+
+        LOG.debug("TestCacheMap.runTests(), Second batch of {} inserts starting from {}", initialCapacity / 2, initialCapacity);
+
+        for (int i = initialCapacity; i < initialCapacity + initialCapacity / 2; i++) {
+            String key   = String.valueOf(i);
+            String value = key;
+
+            LOG.debug("TestCacheMap.runTests(), Inserting into Cache, key:{}, value:{}", key, value);
+
+            testCacheMap.put(key, value);
+
+            LOG.debug("TestCacheMap.runTests(), Cache Size after insert(): {}", testCacheMap.size());
+        }
+
+        Set<String> keySet = testCacheMap.keySet();
+
+        if (LOG.isDebugEnabled()) {
```

Comment:
nit: isDebugEnabled may be removed.

Reply:
The whole block is only for debug logging; hence skipping the block when debug is disabled.